changefeedccl: add enriched_properties option

asg0451 opened the pull request:
(see individual commits)

- Add the `enriched_properties` option which will control which properties are included in the enriched envelope.
- The enriched envelope in json should not include the `payload` wrapper unless the schema is included.

Epic: CRDB-8665
Fixes: #139687
Release note: None

asg0451 committed:
changefeedccl: add enriched_properties option

Add the `enriched_properties` option which will control which
properties are included in the enriched envelope.

Fixes: #139687
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/140992)
<!-- Reviewable:end -->

asg0451 committed:
changefeedccl: fix enriched envelope in json

The enriched envelope in json should not include the `payload`
wrapper unless the schema is included.

Fixes: #139687
Release note: None

aerfrei commented on pkg/ccl/changefeedccl/changefeed_test.go:
```diff
@@ -3851,28 +3851,49 @@ func TestChangefeedEnriched(t *testing.T) {
+					msg = fmt.Sprintf(`%s: [0]->{"payload": {"after": {"a": 0, "b": "dog"}, "op": "c"}}`, topic)
+				}
+
+				assertPayloadsEnvelopeStripTs(t, foo, changefeedbase.OptEnvelopeEnriched, []string{msg})
```
Do we also want to add tests for the error messages if you specify an illegal property?

aerfrei commented on pkg/ccl/changefeedccl/changefeed_test.go:
```diff
@@ -3851,28 +3851,49 @@ func TestChangefeedEnriched(t *testing.T) {
+				sqlDB.Exec(t, `INSERT INTO foo values (0, 'dog')`)
+
+				foo := feed(t, f, fmt.Sprintf(`CREATE CHANGEFEED FOR foo WITH envelope=enriched, enriched_properties='%s'`,
+					strings.Join(tc.enrichedProperties, ",")))
```
nit: Would this test pass if we had `enriched_properties='schema, source'` with a space after the comma? Seems like it might not but it might be more user friendly if it did.

asg0451 commented:
bors cancel

asg0451 commented:
bors r=aerfrei
